dev/tests/static/testsuite/Magento/Test/Integrity/DependencyTest.php

```diff
@@ -342,20 +318,29 @@ private static function getAppCodeDir()
      */
     private static function getTableToPrimaryModuleMap(): array
     {
-        $appCode = self::getAppCodeDir();
         $tableToPrimaryModuleMap = [];
-        foreach (glob($appCode . '/*/*/etc/db_schema_whitelist.json') as $file) {
-            $dbSchemaWhitelist = (array)json_decode(file_get_contents($file));
-            preg_match('|.*/(.*)/(.*)/etc/db_schema_whitelist.json|', $file, $matches);
-            $moduleName = $matches[1] . '\\' . $matches[2];
-            $isStagingModule = (substr_compare($moduleName, 'Staging', -strlen('Staging')) === 0);
+
+        /** @var \Magento\Framework\Component\ComponentFile[] $configs */
+        $configs = Files::init()->getConfigFiles(
+            'db_schema_whitelist.json',
+            [],
+            false,
+            true
+        );
+        foreach ($configs as $file) {
+            $dbSchemaWhitelist = (array)json_decode(file_get_contents($file->getFullPath()));
```

Comment:
https://www.php.net/manual/en/function.json-decode.php
![image](https://user-images.githubusercontent.com/1873745/95345707-b1d36c00-08c3-11eb-9a31-dfed3698ff9d.png)

```suggestion
            $dbSchemaWhitelist = json_decode(file_get_contents($file->getFullPath()), true);
```